### PR TITLE
tweak ingress/ports to use Service NodePort in embedded

### PR DESCRIPTION
**Pull request by dexhorthy**
- create a conditional split to use one of two service objects based on embedded vs existing
- Use a NodePort in embedded cluster
- Update app labels to be more consistent with other projects we publish
- Add an "open app" button to the dashboard so that folks don't need to struggle with the arcane magic needed to make this feature work - they'll have a working example they can iterate on out of the box

Other PRs:

https://github.com/replicatedhq/replicated-starter-kots/pull/67
https://github.com/replicatedhq/replicated-docs/pull/555

### CODE REVIEW

**Review by erquhart — APPROVED** [review]
Awesome improvements!

File: example-configmap.yaml [hunk 1]
```diff
@@ -13,19 +13,15 @@ data:
         <link href="styles.css" rel="stylesheet" type="text/css">
       </head>
       <body>
-        Hello from KOTS.
+        This is an example KOTS application.
         <p>This is text from a user config value: '{{repl ConfigOption "example_default_value"}}' </p>
```
Comment (alexparker215) [inline example-configmap.yaml:17]: Any reason this line has paragraph tags but others don't?

File: example-service.yaml [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx
+  labels:
+    app: nginx
+  annotations:
+    kots.io/when: '{{repl IsKurl }}'
+spec:
+  type: NodePort
```
Comment (alexparker215) [inline example-service.yaml:25]: The PR description says this uses a NodePort for existing cluster, but this looks like it uses ClusterIP for existing cluster and NodePort for embedded.

**Review by alexparker215 — COMMENTED** [review]
One question, one minor comment. Otherwise, lgtm.

File: example-service.yaml (hunk 2 above)
Comment (jdewinne) [reply example-service.yaml:25]: Will this work with the `Application` port?
Our docs mention it should be `ClusterIP`
https://docs.replicated.com/reference/custom-resource-application#ports

Comment (dexhorthy) [reply example-service.yaml:25]: ClusterIP for existing cluster, nodeport for embedded -- @ajp-io I updated the PR description to match. I think those docs might be wrong / only for existing cluster (this used to not work for embedded at all)

File: example-configmap.yaml (hunk 1 above)
Comment (dexhorthy) [reply example-configmap.yaml:17]: no reason. This is some very rough HTML and I'm not fussing about style. Open to iterations on this though.

File: example-service.yaml (hunk 2 above)
Comment (dexhorthy) [reply example-service.yaml:25]: @jdewinne clarified https://github.com/replicatedhq/replicated-docs/pull/596

**Review by jdewinne — APPROVED** [review]

**Pull request merged**